go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -308,7 +308,7 @@ func transformAggregator(ctx *plancontext.PlanningContext, op *operators.Aggrega
 			return nil, vterrors.VT12001(fmt.Sprintf("in scatter query: aggregation function '%s'", sqlparser.String(aggr.Original)))
 		}
 		aggrParam := engine.NewAggregateParam(aggr.OpCode, aggr.ColOffset, aggr.Alias, ctx.VSchema.Environment().CollationEnv())
-		aggrParam.Expr = aggr.Func
+		aggrParam.Func = aggr.Func
```

Comment:
should we update the separator here if not provided, then in the execution engine we do not have to check the separator

Reply:
That's a good idea, I've made the change.